lib/fluent/command/unpacker.rb

```diff
@@ -0,0 +1,253 @@
+
+    def initialize(argv = ARGV)
+      super
+
+      @options.merge!(DEFAULT_OPTIONS)
+      @params = {}
+      if i = @argv.index('--')
+        @params = @argv[i+1..-1].reduce({}) do |acc, e|
+          k, v = e[1..-1].split('=')
+          acc.merge(k => v)
+        end
+        @argv = @argv[0...i]
+      end
+
+      configure_option_parser
+    end
+
+    private
+
+    def configure_option_parser
+      @opt_parser.banner = "Usage: fluent-unpacker #{self.class.to_s.split('::').last.downcase} [options] file [-- <params>]"
+
+      @opt_parser.on('-f TYPE', '--format', 'configure output format') do |v|
+        @options[:format] = v.to_sym
+      end
+    end
+
+    def lookup_formatter(format, params)
+      conf = Fluent::Config::Element.new('ROOT', '', params, [])
+      formatter = Fluent::Plugin.new_formatter(format)
+
+      if formatter.respond_to?(:configure)
+        formatter.configure(conf)
+      end
+      formatter
+    rescue => e
+      usage e
+    end
+  end
+
+  class Head < Base
+    include Formattable
+
+    DEFAULT_HEAD_OPTIONS = {
+      count: 5
+    }
+
+    def initialize(argv = ARGV)
+      super
+      @options.merge!(DEFAULT_HEAD_OPTIONS)
+      @path = parse_options!
+    end
+
+    def call
+      @formatter = lookup_formatter(@options[:format], @params)
+
+      File.open(@path, 'r') do |io|
+        i = 0
+        MessagePack::Unpacker.new(io).each do |(time, record)|
+          break if i == @options[:count]
+          i += 1
+          puts @formatter.format(@path, time, record) # tag is use for tag
+        end
+      end
+    end
+
+    private
+
+    def parse_options!
+      @opt_parser.on('-n COUNT', 'Set the number of lines to display') do |v|
+        @options[:count] = v.to_i
+      end
+
+      path = super
+
+      case
+      when path.empty?
+        usage 'Path is required'
+      when !File.exist?(path.first)
+        usage "#{path.first} is not found"
+      when @options[:count] < 1
+        usage "illegal line count -- #{@options[:count]}"
+      else
+        path.first
+      end
+    end
+  end
+
+  class Cat < Base
+    include Formattable
+
+    def initialize(argv = ARGV)
+      super
+      @path = parse_options!
+    end
+
+    def call
+      @formatter = lookup_formatter(@options[:format], @params)
+
+      File.open(@path, 'r') do |io|
+        MessagePack::Unpacker.new(io).each do |(time, record)|
+          puts @formatter.format(@path, time, record) # @path is used for tag
+        end
+      end
+    end
+
+    def parse_options!
+      path = super
+      usage 'Path is required' if path.empty?
+      usage "#{path.first} is not found" unless File.exist?(path.first)
+      path.first
+    end
+  end
+
+  class Formats < Base
+    def initialize(argv = ARGV)
+      super
+      parse_options!
+    end
+
+    def call
+      prefix = Fluent::Plugin::FORMATTER_REGISTRY.dir_search_prefix || 'formatter_'
+
+      new_path = Fluent::Plugin::FORMATTER_REGISTRY.paths.last
```

Comment:
Is this line for `-p DIR` option?
If so, that value can be fetched by `@options[:plugins]` simply, instead of fetching last item of registry paths.